ddtrace/internal/sampling.py

```diff
@@ -155,7 +148,28 @@ def _sample(self, span):
 
     def match(self, span):
         """Determines if the span's service and name match the configured patterns"""
-        return self._service_matcher.match(span.service) and self._name_matcher.match(span.name)
+        name = span.name
+        service = span.service
+        # If a span lacks a name and service, we can't match on it
+        if service is None and name is None:
+            return False
+
+        # Default to True, as the rule may not have a name or service rule
+        # For whichever rules it does have, it will attempt to match on them
+        service_match = True
+        name_match = True
+
+        if self._service_matcher:
+            if span.service is None:
+                return False
+            else:
+                service_match = self._service_matcher.match(span.service)
+        if self._name_matcher:
+            if span.name is None:
+                return False
+            else:
+                name_match = self._name_matcher.match(span.name)
+        return service_match and name_match
```

Comment:
Would this work?

```suggestion
        service_match = False
        if self._service_matcher and span.service:
                service_match = self._service_matcher.match(span.service)
        name_match = False
        if self._name_matcher and span.name:
                name_match = self._name_matcher.match(span.name)
        return service_match and name_match
```

Reply:
Unfortunately not, because for example if you only set a name pattern e.g `rule = SpanSamplingRule(name="test_name")` then you end up with the service_match being False and the evaluation always being False, when it should be true as long as the span has name="test_name".